cypress/plugins/synapsedocker/index.ts

```diff
@@ -93,21 +105,45 @@ async function cfgDirFromTemplate(template: string): Promise<SynapseConfig> {
     console.log(`Gen ${path.join(templateDir, "localhost.signing.key")}`);
     await fse.writeFile(path.join(tempDir, "localhost.signing.key"), `ed25519 x ${signingKey}`);
 
+    // If we have a `Dockerfile`, this means that the template needs its own image.
+    // If so, let's create a custom Docker image.
+    let matrixImage;
+    if (fse.existsSync(path.join(templateDir, "Dockerfile"))) {
+        matrixImage = `matrix-react-sdk-test-${template}`;
+        try {
+            childProcess.execFileSync('docker', [
+                "build",
```

Comment:
We shouldn't be building ci run images on every ci run. This will expend a lot of time. Also given that our cypress jobs will gain automatic load balancing it'll possibly be done multiple times per run as the parallel runners will be on separate VMs. This slows down tests considerably and can hit dockerhub pull rate limits even. We should be publishing the ci images like we do for other things

Reply:
I suggest some ci which publishes images automatically from a directory within cypress into the github package space which can then be used by the cypress runners in ci

Reply:
I'll try and find out how to do this without breaking local testing.

Note, however, that on my computer, building the images takes about 4-5 seconds, so the impact is much lower than adding even a single test.

Reply:
Through our move to github ci we're trying to deduplicate as much as possible. To not make the same anti patterns we had in buildkite where each job built the app itself and a pr commit would do the same work a handful of times. Waste of compute

For local testing you can build those tags locally also